widgets/src/lib/InferenceWidget/shared/WidgetRecorder/WidgetRecorder.svelte

```diff
@@ -21,13 +22,19 @@
 	});
 
 	async function onClick() {
-		isRecording = !isRecording;
-		if (isRecording) {
-			recorder.start();
-			onRecordStart();
-		} else {
-			const blob = await recorder.stopRecording();
-			onRecordStop(blob);
+		try {
+			isRecording = !isRecording;
+			error = "";
+			if (isRecording) {
+				await recorder.start();
+				onRecordStart();
+			} else {
+				const blob = await recorder.stopRecording();
+				onRecordStop(blob);
+			}
+		} catch (e) {
+			isRecording = false;
+			error = "Please allow 🤗 to access your microphone";
```

Comment:
Maybe check the kind of error here to display a more precise error?

From https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia: "If the user denies permission, or matching media is not available, then the promise is rejected with `NotAllowedError` or `NotFoundError` respectively."